### PR TITLE
Last half of Simple Java examples port

### PR DESCRIPTION
**Pull request by adam47deg**
This PR contains the last half of xef examples ported to Java.
We followed the Kotlin versions as closely as possible:
Clickup ticket link: https://app.clickup.com/t/861mwhzxp

### CODE REVIEW

**Review by Yawolf — APPROVED** [review]

File: examples/java/src/main/java/com/xebia/functional/xef/java/auto/Movies.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    public static class Movie {
+        public String title;
+        public String genre;
+        public String director;
+    }
+
+    public static void main(String[] args) throws ExecutionException, InterruptedException {
+        try (AIScope scope = new AIScope()) {
+            scope.prompt("Please provide a movie title, genre and director for the Inception movie", Movie.class)
+                    .thenAccept((movie) -> System.out.println("The movie " + movie.title + " is a " +
```
Comment (Zevleg) [inline examples/java/src/main/java/com/xebia/functional/xef/java/auto/Movies.java:16]: Remove the parentheses around the "movie" parameter

File: examples/java/src/main/java/com/xebia/functional/xef/java/auto/Persons.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+    public static class Person {
+        public String name;
+        public int age;
+    }
+
+    public static void main(String[] args) throws ExecutionException, InterruptedException {
+        try (AIScope scope = new AIScope()) {
+            scope.prompt("What is your name and age?", Person.class)
+                    .thenAccept((person) -> System.out.println("Hello " + person.name + ", you are " + person.age + " years old."))
```
Comment (Zevleg) [inline examples/java/src/main/java/com/xebia/functional/xef/java/auto/Persons.java:15]: Remove the parentheses around the "person" parameter

File: examples/java/src/main/java/com/xebia/functional/xef/java/auto/Planets.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        try (AIScope scope = new AIScope()) {
+            CompletableFuture<Planet> earth = scope.prompt("Information about Earth and its moon.", Planet.class);
+            CompletableFuture<Planet> mars = scope.prompt("Information about Mars and its moons.", Planet.class);
+
+            System.out.println("Celestial bodies information:\n\n" + planetInfo(earth.get()) + "\n\n" + planetInfo(mars.get()));
+        }
+    }
+
+    private static String planetInfo(Planet planet){
+        List<String> moonList = planet.moons.stream().map(it -> "  - " + it.name + ": " + it.distanceFromPlanetInKm + " km away from " + planet.name).collect(toList());
```
Comment (Zevleg) [inline examples/java/src/main/java/com/xebia/functional/xef/java/auto/Planets.java:31]: Replace this usage of 'Stream.collect(Collectors.toList())' with 'Stream.toList()'

File: examples/java/src/main/java/com/xebia/functional/xef/java/auto/Poems.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+            CompletableFuture<Poem> poem2 = scope.prompt("A short poem about the power of technology.", Poem.class);
+            CompletableFuture<Poem> poem3 = scope.prompt("A short poem about the wisdom of artificial intelligence.", Poem.class);
+
+            String combinedPoems = String.format("%s\n\n%s\n\n%s", poem1.get().content, poem2.get().content, poem3.get().content);
+            String newPoemPrompt = "Write a new poem that combines ideas from the following themes: the beauty " +
+                    "of nature, the power of technology, and the wisdom of artificial intelligence. Here are some " +
+                    "examples of poems on these themes: " + combinedPoems;
+
+            scope.prompt(newPoemPrompt, Poem.class).
+                    thenAccept((poem) -> System.out.printf("New Poem:\n\n" + poem.content))
```
Comment (Zevleg) [inline examples/java/src/main/java/com/xebia/functional/xef/java/auto/Poems.java:24]: Remove the parentheses around the "poem" parameter

**Review by Zevleg — CHANGES_REQUESTED** [review]

File: examples/java/src/main/java/com/xebia/functional/xef/java/auto/Movies.java (hunk 1 above)
Comment (adam47deg) [reply examples/java/src/main/java/com/xebia/functional/xef/java/auto/Movies.java]: Fixed

File: examples/java/src/main/java/com/xebia/functional/xef/java/auto/Persons.java (hunk 2 above)
Comment (adam47deg) [reply examples/java/src/main/java/com/xebia/functional/xef/java/auto/Persons.java]: Fixed

File: examples/java/src/main/java/com/xebia/functional/xef/java/auto/Poems.java (hunk 4 above)
Comment (adam47deg) [reply examples/java/src/main/java/com/xebia/functional/xef/java/auto/Poems.java]: Fixed

File: examples/java/src/main/java/com/xebia/functional/xef/java/auto/TopAttractions.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    static class Weather {
+        public City city;
+        public double temperature;
+        public String description;
+    }
+
+    public static void main(String[] args) throws ExecutionException, InterruptedException {
+        try (AIScope scope = new AIScope()) {
+            scope.prompt("Top attraction in Cádiz, Spain.", TopAttraction.class)
+                .thenAccept((attraction) -> System.out.println(
```
[the remark this answers is not in the document]
Comment (adam47deg) [reply examples/java/src/main/java/com/xebia/functional/xef/java/auto/TopAttractions.java]: Fixed

File: examples/java/src/main/java/com/xebia/functional/xef/java/auto/TouristAttractions.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    static class TouristAttraction {
+        public String name;
+        public String location;
+        public String history;
+    }
+
+    public static void main(String[] args) throws ExecutionException, InterruptedException {
+        try (AIScope scope = new AIScope()) {
+            scope.prompt("Statue of Liberty location and history.", TouristAttraction.class)
+                    .thenAccept((statueOfLiberty) -> System.out.println(
```
[the remark this answers is not in the document]
Comment (adam47deg) [reply examples/java/src/main/java/com/xebia/functional/xef/java/auto/TouristAttractions.java]: Fixed

File: examples/java/src/main/java/com/xebia/functional/xef/java/auto/Movies.java [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+        public String title;
+        public String genre;
+        public String director;
+    }
+
+    public static void main(String[] args) throws ExecutionException, InterruptedException {
+        try (AIScope scope = new AIScope()) {
+            scope.prompt("Please provide a movie title, genre and director for the Inception movie", Movie.class)
+                    .thenAccept(movie -> System.out.println("The movie " + movie.title + " is a " +
+                            movie.genre + " film directed by " + movie.director + "."))
```
Comment (diesalbla) [inline examples/java/src/main/java/com/xebia/functional/xef/java/auto/Movies.java:17]: Perhaps we could extract a method `writeDown(): String` on the `Movie` class?

File: examples/java/src/main/java/com/xebia/functional/xef/java/auto/Persons.java [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+    public static class Person {
+        public String name;
+        public int age;
+    }
+
+    public static void main(String[] args) throws ExecutionException, InterruptedException {
+        try (AIScope scope = new AIScope()) {
+            scope.prompt("What is your name and age?", Person.class)
+                    .thenAccept(person -> System.out.println("Hello " + person.name + ", you are " + person.age + " years old."))
```
Comment (diesalbla) [inline examples/java/src/main/java/com/xebia/functional/xef/java/auto/Persons.java:15]: Likewise, perhaps we could extract a `writeDown` method in `Person` class?

**Review by diesalbla — DISMISSED** [review]
Small comments :)

File: examples/java/src/main/java/com/xebia/functional/xef/java/auto/Movies.java (hunk 7 above)
Comment (adam47deg) [reply examples/java/src/main/java/com/xebia/functional/xef/java/auto/Movies.java]: Good idea, fixed by adding 'toString' on Movie

File: examples/java/src/main/java/com/xebia/functional/xef/java/auto/Persons.java (hunk 8 above)
Comment (adam47deg) [reply examples/java/src/main/java/com/xebia/functional/xef/java/auto/Persons.java]: Good idea, fixed by adding 'toString' on Person

**Review by Zevleg — APPROVED** [review]

**Pull request merged**